engine/src/main/java/org/pentaho/di/trans/steps/scriptvalues_mod/ScriptValuesAddedFunctions.java

```diff
@@ -0,0 +1,2874 @@
+          if ( modulus == 0 ) {
+            returnCode = true;
+          }
+        } catch ( Exception e ) {
+          // No Need to throw exception
+          // This means that input can not be parsed to Integer
+        }
+
+      }
+    } else {
+      throw Context.reportRuntimeError( "The function call LuhnCheck requires 1 argument." );
+
+    }
+    return returnCode;
+  }
+
+  public static int indexOf( Context actualContext, Scriptable actualObject, Object[] ArgList,
+    Function FunctionContext ) {
+
+    int returnIndex = -1;
+
+    if ( ArgList.length == 2 || ArgList.length == 3 ) {
+      if ( !isNull( ArgList ) && !isUndefined( ArgList ) ) {
+        String string = Context.toString( ArgList[0] );
+        String subString = Context.toString( ArgList[1] );
+
+        int fromIndex = 0;
+        if ( ArgList.length == 3 ) {
+          fromIndex = (int) Math.round( Context.toNumber( ArgList[2] ) );
+        }
+        returnIndex = string.indexOf( subString, fromIndex );
+      }
+    } else {
+      throw Context.reportRuntimeError( "The function call indexOf requires 2 or 3 arguments" );
+    }
+    return returnIndex;
+  }
+
+  public static Object getTransformationName( Context actualContext, Scriptable actualObject, Object[] ArgList,
+    Function FunctionContext ) {
+    try {
+      Object objTranName = Context.toString( actualObject.get( "_TransformationName_", actualObject ) );
+      return objTranName;
+    } catch ( Exception e ) {
+      throw Context.reportRuntimeError( e.toString() );
+    }
+  }
+
+  public static void appendToFile( Context actualContext, Scriptable actualObject, Object[] ArgList,
+    Function FunctionContext ) {
+
+    if ( !isNull( ArgList ) && !isUndefined( ArgList ) ) {
+      try {
+        FileOutputStream file = new FileOutputStream( Context.toString( ArgList[0] ), true );
+        DataOutputStream out = new DataOutputStream( file );
+        out.writeBytes( Context.toString( ArgList[1] ) );
+        out.flush();
+        out.close();
+      } catch ( Exception er ) {
+        throw Context.reportRuntimeError( er.toString() );
+      }
+    } else {
+      throw Context.reportRuntimeError( "The function call appendToFile requires arguments." );
+    }
+  }
+
+  public static Object getFiscalDate( Context actualContext, Scriptable actualObject, Object[] ArgList,
+    Function FunctionContext ) {
+
+    if ( ArgList.length == 2 ) {
+      try {
+        if ( isNull( ArgList ) ) {
+          return null;
+        } else if ( isUndefined( ArgList ) ) {
+          return Context.getUndefinedValue();
+        }
+        java.util.Date dIn = (java.util.Date) Context.jsToJava( ArgList[0], java.util.Date.class );
+        Calendar startDate = Calendar.getInstance();
+        Calendar fisStartDate = Calendar.getInstance();
+        Calendar fisOffsetDate = Calendar.getInstance();
+        startDate.setTime( dIn );
+        Format dfFormatter = new SimpleDateFormat( "dd.MM.yyyy" );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.